packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -216,10 +236,17 @@ export class Metric implements IMetric {
     if (periodSec !== 1 && periodSec !== 5 && periodSec !== 10 && periodSec !== 30 && periodSec % 60 !== 0) {
       throw new Error(`'period' must be 1, 5, 10, 30, or a multiple of 60 seconds, received ${periodSec}`);
     }
+    if (props.dimensions && props.dimensionsMap) {
+      throw new Error('Using both the \'dimensions\' and \'dimensionsMap\' properties is not supported. Use only \'dimensionsMap.\'');
+    }
     if (props.dimensions) {
       this.validateDimensions(props.dimensions);
     }
     this.dimensions = props.dimensions;
```

Comment:
In my opinion, `dimensions` should also be populated if only `props. dimensionsMap` was provided, to ease the transition for the customers.